pkg/apis/core/validation/validation.go

```diff
@@ -1920,7 +1913,7 @@ func ValidatePersistentVolumeClaimUpdate(newPvc, oldPvc *core.PersistentVolumeCl
 		}
 	}
 
-	if utilfeature.DefaultFeatureGate.Enabled(features.BlockVolume) {
+	if oldPvc.Spec.VolumeMode != nil {
```

Comment:
why is this only checked if the old volumemode is non-nil? is changing from nil to non-nil allowed? 

also, I expect this introduces a similar failure mode as the unconditional immutable check for PV

Reply:
Right, I do see these two failures:
```console
=== RUN   TestSetDefaultPersistentVolume
--- FAIL: TestSetDefaultPersistentVolume (0.00s)
    /Users/sbezverk/go/src/k8s.io/kubernetes/pkg/apis/core/v1/defaults_test.go:824: Expected VolumeMode to not be defaulted, got: 0xc0001e9380
=== RUN   TestSetDefaultPersistentVolumeClaim
--- FAIL: TestSetDefaultPersistentVolumeClaim (0.00s)
    /Users/sbezverk/go/src/k8s.io/kubernetes/pkg/apis/core/v1/defaults_test.go:853: Expected VolumeMode to not be defaulted, got: 0xc00007da80
```
Should I refactor ValidatePersistentVolumeClaimUpdate and ValidatePersistentVolumeUpdate to follow outlined cases?

Reply:
Yes, making the default unconditional requires updating the unit tests

Reply:
It looks like the common key for all cases is that old and new must be identical. 
in this case I see 3 test cases:
1. old nil - new defaulting to filesystem
2. old filesystem - new filesystem
3. old block volume - new block volume
Would you agree?

Reply:
once the defaulting is unconditional, ValidatePersistentVolumeClaimUpdate and ValidatePersistentVolumeUpdate should never see an object with a nil volumeMode field

Reply:
I refactored `TestSetDefaultPersistentVolumeClaim` in defaults_test.go, please let me know if you are ok with this approach, then I will do the same for pv.